components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -0,0 +1,348 @@
+                                                authzUser.getAuthenticatedSubjectIdentifier(), tenantDomain, username);
+                            } catch (IdentityOAuth2Exception e) {
+                                log.error("Error while revoking access tokens related for the application " + appDTO
+                                        .getApplicationName() + "and the for user " + authenticatedUser.getUserName()
+                                        + " : " + e);
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Method to retrieve applications authorized for user
+     * @param authenticatedUser authenticated user info
+     * @return array of authorized applications
+     * @throws IdentityOAuthAdminException exception
+     */
+    private OAuthConsumerAppDTO[] getAppsAuthorizedByUser(AuthenticatedUser authenticatedUser)
+            throws IdentityOAuthAdminException {
+
+        OAuthAppDAO appDAO = new OAuthAppDAO();
+
+        String tenantAwareusername = authenticatedUser.getUserName();
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        String username = UserCoreUtil.addTenantDomainToEntry(tenantAwareusername, tenantDomain);
+
+        String userStoreDomain = authenticatedUser.getUserStoreDomain();
+
+        Set<String> clientIds;
+        try {
+            clientIds = OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO()
+                    .getAllTimeAuthorizedClientIds(authenticatedUser);
+        } catch (IdentityOAuth2Exception e) {
+            String errorMsg = "Error occurred while retrieving apps authorized by User ID : " + username;
+            throw handleError(errorMsg, e);
+        }
+
+        Set<OAuthConsumerAppDTO> appDTOs = new HashSet<>();
+        for (String clientId : clientIds) {
+            Set<AccessTokenDO> accessTokenDOs;
+            try {
+                accessTokenDOs = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                        .getAccessTokens(clientId, authenticatedUser, userStoreDomain, true);
+            } catch (IdentityOAuth2Exception e) {
+                String errorMsg =
+                        "Error occurred while retrieving access tokens issued for " + "Client ID : " + clientId
+                                + ", User ID : " + username;
+                throw handleError(errorMsg, e);
+            }
+            if (!accessTokenDOs.isEmpty()) {
+                Set<String> distinctClientUserScopeCombo = new HashSet<>();
+                for (AccessTokenDO accessTokenDO : accessTokenDOs) {
+                    AccessTokenDO scopedToken;
+                    String scopeString = OAuth2Util.buildScopeString(accessTokenDO.getScope());
+                    try {
+                        scopedToken = OAuthTokenPersistenceFactory.getInstance().
```

Comment:
Don't we need to check OAuthTokenPersistenceFactory.getInstance() != null here?

Reply:
It's getting handled through catch block